src/main/scala/io/gatling/build/compile/GatlingCompilerSettingsPlugin.scala

```diff
@@ -25,44 +25,68 @@ object GatlingCompilerSettingsPlugin extends AutoPlugin {
 
   override def trigger = allRequirements
 
+  trait GatlingCompilerSettingsKeys {
+    val javaReleaseVersion = settingKey[String]("Java release releaseVersion")
+  }
+
+  object GatlingCompilerSettingsKeys extends GatlingCompilerSettingsKeys
+  object autoImport extends GatlingCompilerSettingsKeys
+
+  import autoImport._
+
+  private val DefaultJavaReleaseVersion = "1.8"
   private val isJava8 = scala.util.Properties.javaVersion.startsWith("1.8")
 
-  private val JavacOptions =
-    if (isJava8) {
-      Seq("-source", "1.8", "-target", "1.8")
+  private val JavacOptions = Def.setting {
+    val version = javaReleaseVersion.value
+    if (version == DefaultJavaReleaseVersion) {
+      if (isJava8) {
+        Seq("-source", "1.8", "-target", "1.8")
+      } else {
+        Seq("--release", "1.8")
+      }
     } else {
-      Seq("--release", "8")
+      // When user overwrite java release version, we leave him the responsibility
+      Seq.empty
     }
+  }
 
-  private val ScalacOptions = {
+  private val ScalacOptions = Def.setting {
+    val releaseVersion = javaReleaseVersion.value
     val base =
       Seq(
         "-encoding",
         "UTF-8",
         "-deprecation",
         "-feature",
         "-unchecked",
-        "-language:implicitConversions",
-        "-target:jvm-1.8"
+        "-language:implicitConversions"
       )
 
-    if (isJava8) {
+    if (releaseVersion != DefaultJavaReleaseVersion) {
+      // When user overwrite java release version, we leave him the responsibility
       base
     } else {
-      base ++ Seq("-release", "8")
+      val baseWithTarget = base :+ "-target:jvm-1.8"
+      if (isJava8) {
+        baseWithTarget
+      } else {
+        baseWithTarget ++ Seq("-release", "8")
+      }
     }
```

Comment:
I personally, would have create a

```scala
gatlingScalacJavaVersion := if (isJava8) {
  Seq("-target:jvm-1.8")
} else {
  Seq("-target:jvm-1.8", "-target", "8")
}
```

And adding this settings to the `scalacOptions`
That allows plugin users to only override that.

Reply:
?! Scalac is very likely to either choke or only honor the last value. Is this something you've tested?